### PR TITLE
Remove detector weight decay

### PR DESCRIPTION
**Pull request by guarin**
## What has changed and why?

* Remove `detector_weight_decay` argument from LTDETR
* Remove LTDETR default optimizer param group
* Raise error if params require grad but are not in any optimizer group

This argument is not really needed and was a small bug. The default in RTDETRv2 is to apply the same weight decay to the backbone and the detector with the exception of the norm and bias layers which do not use weight decay. This PR fixes an issue where we mistakenly applied no weight decay to all layers in the detector.

## How has it been tested?

* Manually

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Did you update the documentation?

- [ ] Yes
- [x] Not needed (internal change without effects for user)

### CODE REVIEW

File: src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py [hunk 1]
```diff
@@ -114,8 +114,6 @@ class DINOv2LTDETRObjectDetectionTrainArgs(TrainModelArgs):
     backbone_lr_factor: float = 1e-2
     backbone_weight_decay: float | None = None  # Use default if None
```
[the remark this answers is not in the document]
Comment (guarin) [reply src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py]: This is ok

File: src/lightly_train/_commands/train_task_helpers.py [hunk 2]
```diff
@@ -1394,3 +1395,43 @@ def finetune_from_checkpoint(
...
+
+    # Log optimizer parameter groups
+    param_id_to_name = {id(param): name for name, param in model.named_parameters()}
+    logger.debug("Optimizer parameter groups")
+    for group in optimizer.param_groups:
+        group_info = group.copy()
+        group_info.pop("params", None)
+        logger.debug(f"  {group_info}")
+        for param in group["params"]:
+            param_name = param_id_to_name[id(param)]
```
Comment (yutong-xiang-97) [inline src/lightly_train/_commands/train_task_helpers.py:1421]: Is there a potential to get a `KeyError` here?

File: src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py [hunk 3]
```diff
@@ -114,8 +114,6 @@ class DINOv2LTDETRObjectDetectionTrainArgs(TrainModelArgs):
...
 
-    detector_weight_decay: float = 0.0
```
Comment (yutong-xiang-97) [inline src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py:118]: This is a breaking change for anyone with an existing YAML config (from Codex):

Removing this public model arg turns previously valid LT-DETR training configs into hard failures. `TrainModelArgs` inherits the repo-wide      
strict config behavior `(extra="forbid")`, so any saved YAML/CLI config that still sets `model_args.detector_weight_decay` (often explicitly to 0.0, the old default) will now fail validation before training starts.

File: src/lightly_train/_task_models/dinov3_ltdetr_object_detection/train_model.py [hunk 4]
```diff
@@ -113,8 +113,6 @@ class DINOv3LTDETRObjectDetectionTrainArgs(TrainModelArgs):
     backbone_lr_factor: float = 1e-2
     backbone_weight_decay: float | None = None  # Use default if None
 
-    detector_weight_decay: float = 0.0
```
Comment (yutong-xiang-97) [inline src/lightly_train/_task_models/dinov3_ltdetr_object_detection/train_model.py:117]: Same here

**Review by yutong-xiang-97 — COMMENTED** [review]
Left some comments.

A clarifying question: by "The default in RTDETRv2 is to apply the same weight decay to the backbone and the detector with the exception of the norm and bias layers which do not use weight decay. ", do you mean that both the backbone and detector use the identical weight decay (minus norm and bias layers), or all the backbone / detector parameters uses the same weight decay, but weight decay values can be differenr for backbone and detector?

File: src/lightly_train/_commands/train_task_helpers.py (hunk 2 above)
Comment (guarin) [reply src/lightly_train/_commands/train_task_helpers.py:1421]: Theoretically no. `model` contains all possible parameters that should be in the optimizer param groups. In practice, if there is some other parameter in there we definitely want it to raise a KeyError

File: src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py (hunk 3 above)
Comment (guarin) [reply src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py:118]: Yes and this is fine as this parameter should really not be used.

**Comment by guarin** [comment]
> A clarifying question: by "The default in RTDETRv2 is to apply the same weight decay to the backbone and the detector with the exception of the norm and bias layers which do not use weight decay. ", do you mean that both the backbone and detector use the identical weight decay (minus norm and bias layers), or all the backbone / detector parameters uses the same weight decay, but weight decay values can be differenr for backbone and detector?

They use the same weight decay values

File: src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py (hunk 3 above)
Comment (yutong-xiang-97) [reply src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py:118]: Then should we give an heads-up in the Changelog or something? My understanding is that users have their own YAML files, and this would give an error once they update the package.

**Comment by yutong-xiang-97** [comment]
> > A clarifying question: by "The default in RTDETRv2 is to apply the same weight decay to the backbone and the detector with the exception of the norm and bias layers which do not use weight decay. ", do you mean that both the backbone and detector use the identical weight decay (minus norm and bias layers), or all the backbone / detector parameters uses the same weight decay, but weight decay values can be differenr for backbone and detector?
> 
> They use the same weight decay values

OK but now backbone uses `backbone_weight_decay` if there is a `self.model_args.backbone_weight_decay` and fall back to `self.model_args.weight_decay` while detector uses `self.model_args.weight_decay`. Is this intended?

**Comment by guarin** [comment]
> OK but now backbone uses backbone_weight_decay if there is a self.model_args.backbone_weight_decay and fall back to self.model_args.weight_decay while detector uses self.model_args.weight_decay. Is this intended?

Ah yes, great point. I'll also remove the `backbone_weight_decay`

File: src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py (hunk 3 above)
Comment (guarin) [reply src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py:118]: Updated the changelog. I don't expect that anyone used these arguments (`detector_weight_decay` and `backbone_weight_decy`)
